errors.go

```diff
@@ -36,7 +36,7 @@ var (
 	errCompressionMethodUnset           = &FatalError{errors.New("server hello can not be created without a compression method")}
 	errCookieMismatch                   = &FatalError{errors.New("client+server cookie does not match")}
 	errCookieTooLong                    = &FatalError{errors.New("cookie must not be longer then 255 bytes")}
-	errHandshakeTimeout                 = &FatalError{xerrors.Errorf("the connection timed out during the handshake: %w", context.DeadlineExceeded)}
+	errHandshakeTimeout                 = &FatalError{&HandshakeError{xerrors.Errorf("the connection timed out during the handshake: %w", context.DeadlineExceeded)}}
```

Comment:
I don't have strong opinion, but wrapping at https://github.com/pion/dtls/pull/258/files#diff-a2ddca1600f34abb91244424f8431c2cL889 as same as the other errors would have better uniformity.

Reply:
It'd be nice, but whichever way I go there, it'll look inconsistent. The code-path there used to return a FatalError and I didn't want to change that while getting a HandshakeError inside it. Could move that piece of code but it'll look equally out of place. Any suggestions of how to arrange it better without changing the semantics?

Reply:
Other handshake errors except `errHandshakeTimeout` are now `&HandshakeError{&FatalError{error}}` https://github.com/pion/dtls/pull/258/files#diff-a2ddca1600f34abb91244424f8431c2cR887-R892.
Only `errHandshakeTimeout` is `&FatalError{&HandshakeError{error}}`.
I think it's better to unify them.

Reply:
Ideally I'd like to change it to something like this:
```go
func (c *Conn) translateHandshakeCtxError(err error) error {
	if err == nil {
		return nil
	}
	if errors.Is(err, context.Canceled) && c.isHandshakeCompletedSuccessfully() {
		return nil
	}
	return &HandshakeError{err}
}
```
This looks correct to me since it returns a HandshakeError when a handshake fails which can be temporary or permanent (depending on what error wrapped). However that'll mean Server(), Dial(), Client() do no longer always return a FatalError like currently. If that works for you, I can make the necessary changes. Alternatively, I can start wrapping every error that comes out of Server(), Dial(), etc in a FatalError in their respective functions to keep the current behavior (which doesn't seem quite right).
To me, FatalError means "this operation failed and there's no hope this could ever work" which is correct for things like bad config, certs and the like. A timeout in the handshake doesn't really fall into this category, even for a Dial() operation.

Reply:
IMO, `FatalError` is not explicitly telling `operation is failed`, but just both `.Temporary()` and `.Timeout()` (that implements `net.Error`) return false.
If returning `&HandshakeError{context.DeadlineExceeded}` without using `errHandshakeTimeout`, the error doesn't implement `net.Error` and user can't know it is a timeout error by unified way.

BTW, `errHandshakeTimeout` should be `TimeoutError` error instead of `FatalError`.

Reply:
Thanks. The HandshakeError implements net.Error now and passes Temporary() and Timeout() to the inner error.